File: python310/task_4.py
def calculate_clock_angle(hours, minutes):
    if (0 <= hours <= 12) and (0 <= minutes <= 59):
        angle = abs(30 * hours - (11 / 2) * minutes)

        if angle > 180:
            angle = 360 - angle

        return angle
    else:
        raise ValueError("Неправильный формат времени")


hours = int(input("Часы: "))
minutes = int(input("Минуты: "))
angle = calculate_clock_angle(hours, minutes)
print(f"Угол между стрелками в {hours}:{minutes} = {angle} градусов")

File: JS/task_4.js
export const task4 = 'Задача 4.  Напишите метод, который будет вычислять угол между часовой и минутной стрелками ' +
    'часов. На вход функции подаётся время в виде двух переменных: "hours" и "minutes".' +
    'Расчет угла производить относительно реального поведения стрелок часов.'

export function calculateClockAngle(hours, minutes) {
    if ((hours >= 0 && hours <= 12) && (minutes >= 0 && minutes <= 59)) {
        let angle = Math.abs(30 * hours - (11 / 2) * minutes);
        if (angle > 180) {
            angle = 360 - angle;
        }
        return angle;
    } else {
        throw new Error("Неправильный формат времени");
    }
}